packages/engine/src/worker/runner/javascript/mod.rs

```diff
@@ -70,175 +74,321 @@ fn get_pkg_path(name: &str, pkg_type: PackageType) -> String {
+                    ]
+                }})",
+            ),
+        )?;
+
+        let pkg: v8::Local<'_, v8::Function> = {
+            let mut try_catch = v8::TryCatch::new(scope);
+
+            let pkg = v8::Script::compile(&mut try_catch, wrapped_code, None).ok_or_else(|| {
+                let exception = try_catch.exception().unwrap();
+                let exception_string = exception
+                    .to_string(&mut try_catch)
+                    .unwrap()
+                    .to_rust_string_lossy(&mut try_catch);
+
+                Error::PackageImport(
+                    path.clone(),
+                    format!("Could not compile code for package: {exception_string}"),
+                )
+            })?;
+
+            let pkg = pkg.run(&mut try_catch).ok_or_else(|| {
+                let exception = try_catch.exception().unwrap();
+                let exception_string = exception
+                    .to_string(&mut try_catch)
+                    .unwrap()
+                    .to_rust_string_lossy(&mut try_catch);
+
+                Error::PackageImport(
+                    path.clone(),
+                    format!("Couldn't execute package setup: {exception_string}"),
+                )
+            })?;
+
+            pkg.try_into().map_err(|err| {
+                Error::PackageImport(
+                    path.clone(),
+                    format!("Could not convert package setup return value into Function: {err}"),
+                )
+            })?
+        };
+
+        let args = &[embedded.hash_util, embedded.hash_stdlib];
+        let fns: v8::Local<'_, v8::Array> = {
+            let global_context = context.global(scope);
+            let fns = call_js_function(scope, pkg, global_context.into(), args).map_err(|err| {
+                Error::PackageImport(path.clone(), format!("Couldn't call package: {err}"))
+            })?;
+
+            fns.try_into().map_err(|err| {
+                Error::PackageImport(
+                    path.clone(),
+                    format!(
+                        "Couldn't convert package return value to array of package functions: \
+                         {err}"
+                    ),
+                )
+            })?
+        };
+
+        if fns.length() != 3 {
+            return Err(Error::PackageImport(
+                path.clone(),
+                "Unexpected amount of returned arguments".into(),
+            ));
         }
 
         // Validate returned array.
         let fn_names = ["start_experiment", "start_sim", "run_task"];
-        for (elem, fn_name) in fns.clone().elements().zip(fn_names) {
-            let elem: mv8::Value<'_> = elem.map_err(|e| {
-                Error::PackageImport(path.clone(), format!("Couldn't index array: {:?}", e))
-            })?;
+        for (idx_fn, fn_name) in fn_names.iter().enumerate() {
+            let elem: v8::Local<'_, v8::Value> =
```

Comment:
```suggestion
            let elem: Value<'s> =
```

Reply:
Addressed in https://github.com/hashintel/hash/pull/414/commits/f4c4f5dc5b872909afc91ff48c1042a1fc74ae2d